### PR TITLE
SAK-50666 Samigo prevent opening links from SEB by default

### PR DESCRIPTION
**Pull request by st-manu**
https://sakaiproject.atlassian.net/browse/SAK-50666

On the Sakai login screen, there are often external links (e.g., to the APAREO page). When accessing the SEB (Safe Exam Browser), this screen appears, allowing users to open links and navigate to Instagram, Twitter, Google, etc. A student could also type a link (e.g., Google) into a short-answer question and open a new tab.

By default, block access to any link outside the platform itself. If a teacher wants to allow specific links, they can configure them using the SEB app.

### CODE REVIEW

**Comment by ottenhoff** [comment]
> If a teacher wants to allow specific links, they can configure them using the SEB app.

Why not allow **and** block using SEB and not Sakai? The approach in this PR just seems opposite of how I think other lockdown browsers work.

If this really does need to go into Sakai, please use a more-specific Sakai property that refers to SEB

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/business/entity/SebConfig.java [hunk 1]
```diff
@@ -92,6 +97,8 @@ public class SebConfig {
     public static final String USER_CONFIRM_QUIT = "ASSESSMENT_SEB_USER_CONFIRM_QUIT";
 
     private static final String QUIT_PASSWORD = SecureDeliveryServiceAPI.EXITPWD_KEY;
+    
+    private ServerConfigurationService serverConfigurationService = ComponentManager.get(ServerConfigurationService.class);
```
Comment (ern) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/business/entity/SebConfig.java:101]: move this init into the constructor

**Comment by st-manu** [comment]
At my university, instructors have been using SEB for some time. It’s intended to create a secure environment where students can only access the exam, and nothing else. Instructors, who use Sakai daily, have expressed frustration about having to use the SEB configuration app for something as basic as preventing navigation to external links. That’s why, with this property, institutions that want to disable external link navigation can enable this feature to apply by default.

I'm not sure if this is what you're referring to @ottenhoff . I'll update the commit tomorrow to add a 'seb.' prefix since I forgot to include it.

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties [hunk 2]
```diff
@@ -3703,6 +3703,14 @@
 # DEFAULT: https://safeexambrowser.org/download_en.html
 # seb.download.link=
 
+# S2U-21 Enable URL filtering for Safe Exam browser
+# DEFAULT: false
+# URL.filter.enable=true
+
+# S2U-21 List of URLs to filter for Safe Exam browser
+# DEFAULT: localhost:8080
```
Comment (ern) [inline config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties:3711]: shouldn't default be SCS.getServerUrl?

**Comment by bgarciaentornos** [comment]
@ottenhoff I think you misunderstood this, Sakai configures what will be applied to SEB, but that happens during the creation of an assessment in Samigo.

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/business/entity/SebConfig.java [hunk 3]
```diff
@@ -264,4 +299,19 @@ private HashMap<String, Object> toMap() {
...
+        return rule;
+    }
+
+    private Map<String, Object> sortMapByKey(Map<String, Object> map) {
+        return map.entrySet().stream()
+                .filter(entry -> entry.getValue() != null)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e1, () -> new TreeMap<>(String.CASE_INSENSITIVE_ORDER)));
+    }
+
+}
```
Comment (bgarciaentornos) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/business/entity/SebConfig.java:317]:
```suggestion
}

```

**Pull request merged**